server/lib/src/plugins/namehistory.rs

```diff
@@ -0,0 +1,268 @@
+use std::sync::Arc;
+
+use kanidm_proto::v1::OperationError;
+
+use crate::entry::{EntryInvalidCommitted, EntrySealedCommitted};
+use crate::event::ModifyEvent;
+use crate::plugins::Plugin;
+use crate::prelude::*;
+use crate::prelude::{BatchModifyEvent, QueryServerWriteTransaction};
+use crate::repl::cid::Cid;
+use crate::value::PartialValue;
+
+pub struct NameHistory {}
+
+lazy_static! {
+    // it contains all the partialvalues used to match against an Entry's class,
+    // we just need a partialvalue to match in order to target the entry
+    static ref CLASSES_TO_UPDATE: [PartialValue; 1] = [PartialValue::new_iutf8("account")];
+    static ref HISTORY_ATTRIBUTES: [&'static str;1] = ["name"];
+}
+
+impl NameHistory {
+    fn is_entry_to_update<VALUE, STATE>(entry: &mut Entry<VALUE, STATE>) -> bool {
+        CLASSES_TO_UPDATE
+            .iter()
+            .any(|pv| entry.attribute_equality("class", pv))
+    }
+
+    fn get_ava_name(history_attr: &str) -> String {
+        format!("{}_history", history_attr)
+    }
+
+    fn handle_name_updates(
+        pre_cand: &[Arc<EntrySealedCommitted>],
+        cand: &mut Vec<EntryInvalidCommitted>,
+        cid: &Cid,
+    ) -> Result<(), OperationError> {
+        for (pre, post) in pre_cand.iter().zip(cand) {
+            // here we check if the current entry has at least one of the classes we intend to target
```

Comment:
Nah, this is the right way to do it :) feel free to remove this comment.

Reply:
Sure!

Reply:
Also resolve these github conversations once youve actioned them, it makes it easier to see what you've done/havent :)

Reply:
Wow I never noticed the resolve button :p